docs/src/man/extending_ls.md

```diff
@@ -1,3 +1,60 @@
 # Extending LiveServer
 
-TBA
+There may be circumstances where you will want the page-reloading to be triggered by your own mechanism.
+As a very simple example, you may want to display your own custom messages every time a file is updated.
+This page explains how to extend `SimpleWatcher <: FileWatcher` and, more generally, how to write your own `FileWatcher`
+
+## Using `SimpleWatcher` with a custom callback
+
+In most circumstances, using an instance of the `SimpleWatcher` type with your own callback function is what you will want to do.
+
+The `SimpleWatcher` does what you expect in terms of watching files it is told to watch and triggering a function (the "callback") when a change is detected.
+The callback function takes as argument the path of the file that was modified and returns nothing.
+
+The base callback function ([`LiveServer.file_changed_callback`](@ref)) does only one thing: it sends a signal to the relevant viewers so that page reloads can be triggered.
+You will typically want to re-use `file_changed_callback` or copy its code.
+
+As an example of a custom callback, here is a simple modified callback mechanism which prints `Hello!` before using the base callback function:
+
+```julia
+custom_callback(fp::AbstractString) = (println("Hello!"); file_changed_callback(fp))
+```
+
+A (slightly) more sophisticated customised callback is the one that is used in [`servedocs`](@ref) which checks if the file that was changed is a `.md` file in which case it triggers a pass of `Documenter.jl` before signalling to current viewers.
+
+## Writing your own `FileWatcher`
+
+If you decide to write your own `FileWatcher` type, you will need to meet the API.
+The easier is probably that you look at the code for `SimpleWatcher` and adapt to your need.
+Let's assume for now that you want to define `CustomWatcher <: FileWatcher`.
+
+### Fields
+
+Your `CustomWatcher` will need to have at least the same fields as `SimpleWatcher` i.e.:
```

Comment:
It depends. If someone implements custom methods for all API functions, he could use whatever fields he wants, right? I don't think this would make sense though. How do you plan to implement the watcher in `JuDoc`?

Reply:
yeah that's true. In JuDoc I found it easier to keep the simplewatcher and modify the coreloop function actually.

I'll modify the doc to specify that if the person only want to modify some of the functions then they will need to keep the corresponding fields and that the methods that must be there (because called by server) are:

* `start` and `stop`
* `watch_file!`
* the `status` field

and that's it